File: app/src/main/java/com/medavox/diabeticdiary/writers/SqliteWriter.java
package com.medavox.diabeticdiary.writers;

import android.content.ContentValues;
import android.content.Context;
import android.util.Log;

import com.medavox.diabeticdiary.db.EntryDatabase;

import java.util.Arrays;

/**
 * @author Adam Howard
 * @date 28/07/2017
 */

public class SqliteWriter implements DataSink {
    private static final String TAG = "SqliteWriter";
    @Override
    public boolean write(Context c, long time, String[] dataValues) {
        for(int i = 0; i < dataValues.length; i++) {
            if(dataValues[i] != null) {
                ContentValues vals = new ContentValues(3);
                vals.put(EntryDatabase.COLUMN_TIME, time);

                vals.put(EntryDatabase.COLUMN_DATA, dataValues[i]);
                vals.put(EntryDatabase.COLUMN_DATA_TYPE, EntryDatabase.dataTypes[i]);
                EntryDatabase.getWritableDB()
                        .insertOrThrow(EntryDatabase.TABLE_NAME
                                , null, vals);
                Log.i(TAG, "wrote to sqlite db: ["+ vals+"]");
            }
        }
        return true;
    }
}

File: app/src/main/java/com/medavox/diabeticdiary/db/SqliteWriter.kt
package com.medavox.diabeticdiary.db

import android.content.Context
import android.content.SharedPreferences
import com.medavox.diabeticdiary.DiabApp
import com.medavox.diabeticdiary.writers.DataSink

/**
 * @author Adam Howard
@date 16/04/2019
 */
class SqliteWriter : DataSink {
    override fun write(sp: SharedPreferences, time: Long, dataValues: Map<EntryType, String>): Boolean {
        DiabApp.dbWorker.post {
            val entris = dataValues.map { Entry(time, it.key, it.value) }
            DiabApp.db().insertEntries(entris)
        }
        return true
    }
}